sympy/integrals/integrals.py

```diff
@@ -428,67 +428,83 @@ def doit(self, **hints):
-            # methods.
-            if meijerg1 is False and meijerg is True:
-                antideriv = None
+                    antideriv = self._eval_integral(
+                        function, xab[0],
+                        meijerg=meijerg, risch=risch, manual=manual,
+                        conds=conds)
             else:
-                antideriv = self._eval_integral(
-                    function, xab[0],
-                    meijerg=meijerg1, risch=risch, manual=manual,
-                    conds=conds)
-                if antideriv is None and meijerg1 is True:
+                '''
+There are a number of tradeoffs in using the Meijer G
+method. It can sometimes be a lot faster than other methods,
+and sometimes slower. And there are certain types of
+integrals for which it is more likely to work than others.
+These heuristics are incorporated in deciding what
+integration methods to try, in what order. See the
+integrate() docstring for details.'''
+                def try_meijerg(function, xab):
+                    ret = None
+                    if len(xab) == 3 and meijerg is not False:
+                        x, a, b = xab
+                        try:
+                            res = meijerint_definite(function, x, a, b)
+                        except NotImplementedError:
+                            from sympy.integrals.meijerint import _debug
+                            _debug('NotImplementedError '
+                                'from meijerint_definite')
+                            res = None
+                        if res is not None:
+                            f, cond = res
+                            if conds == 'piecewise':
+                                ret = Piecewise(
+                                    (f, cond),
+                                    (self.func(
+                                    function, (x, a, b)), True))
+                            elif conds == 'separate':
+                                if len(self.limits) != 1:
+                                    raise ValueError(filldedent('''
+                                        conds=separate not supported in
+                                        multiple integrals'''))
+                                ret = f, cond
+                            else:
+                                ret = f
+                    return ret
+
+                meijerg1 = meijerg
+                if (len(xab) == 3 and xab[1].is_real and xab[2].is_real
+                        and not function.is_Poly and
+                        (xab[1].has(oo, -oo) or xab[2].has(oo, -oo))):
                     ret = try_meijerg(function, xab)
                     if ret is not None:
                         function = ret
                         continue
+                    else:
+                        meijerg1 = False
+                        '''
+If the special meijerg code did not succeed in finding a definite
```

Comment:
Why are these strings here? If this is intended to be a comment, it should be a comment, and indented along with the code block.